nix/deps/clojure/deps.list

```diff
@@ -8,14 +9,17 @@ borkdude/edamame/1.3.23/edamame-1.3.23.jar
 borkdude/sci.impl.reflector/0.0.1/sci.impl.reflector-0.0.1.jar
 camel-snake-kebab/camel-snake-kebab/0.4.3/camel-snake-kebab-0.4.3.jar
 cheshire/cheshire/5.11.0/cheshire-5.11.0.jar
-cider/cider-nrepl/0.44.0/cider-nrepl-0.44.0.jar
-cider/orchard/0.21.0/orchard-0.21.0.jar
-cider/piggieback/0.5.2/piggieback-0.5.2.jar
+cider/cider-nrepl/0.31.0/cider-nrepl-0.31.0.jar
```

Comment:
@siddarthkay, @jakubgs, @yqrashawn: some dependencies keep getting changed because they weren't supposed to be tracked in the file `shadow-cljs.edn`. They are not in general in Clojure projects.

The problematic deps are:

- `cider/cider-nrepl`
- `cider/piggieback`
- `refactor-nrepl`
- `nrepl/nrepl`

The reason they change (especially the first 2) is because `cider/cider-nrepl` and `cider/piggieback` are dependencies for Emacs and we configure specific versions in our machines at `~/.shadow-cljs/config.edn`. Then, when we run `nix-update-clojure`, the deps will actually use our local versions, not the project ones. The ones we have in the project's `shadow-cljs.edn` are very old and probably useless as well.

Here you see the version going down from `0.44.0` to `0.31.0` because I temporarily renamed the path `~/.shadow-cljs`, so that `nix-update-clojure` would pick up the `status-mobile` versions instead.

But I doubt I'll remember to do it in the future or that other devs will do the same, and so we keep perpetually changing these lib versions based on local dev needs...

-----

Just removing the deps from `shadow-cljs.edn` won't do it because our nix script will still see all dependencies managed by shadow that we need to define outside `status-mobile`. One reasonable solution is to filter them out explicitly in the script because it's not like we have a large list of dependencies that keep changing, mostly the ones I shared.

@yqrashawn, do you also customize some of these deps outside `status-mobile`?

Reply:
I do not have these pkgs in my [`~/.shadow-cljs/config.edn`](https://github.com/yqrashawn/yqdotfiles/blob/master/modules/yqrashawn/home-manager/dotfiles/shadow-cljs.edn)

Yes, I use a [custom babashka script](https://github.com/yqrashawn/yqdotfiles/blob/master/modules/yqrashawn/home-manager/dotfiles/local-bins/sstart) that reads these pkgs version using `emacsclient --eval`
and I use `.envrc.local` to `sed` all `yarn shadow-cljs` in `Makefile` to the bb script

```sh
~/local/bin/sstart server
> npx shadow-cljs -d cider/cider-nrepl:0.47.0 -d nrepl/nrepl:1.1.1 -d refactor-nrepl/refactor-nrepl:3.10.0 server
```

Reply:
Thanks @yqrashawn. So it's good to see you also do something about these dependencies because it probably means a quick fix is valuable/worth our time since we don't want every Emacs user to have to figure this sort of thing by themselves.

Reply:
I'll create an issue in any case.

Reply:
Thanks @ilmotta

Reply:
Created https://github.com/status-im/status-mobile/issues/20324